src/init.lua

```diff
@@ -2431,13 +2427,17 @@ end
 xplr.fn.builtin.fmt_general_table_row_cols_1 = function(m)
   local r = m.tree .. m.prefix
 
+  local function path_escape(path)
```

Comment:
Selection list is rendered by Rust code, so the escape logic is there.

I don't think it should be a separate field, since different people may have different opinions of how to escape newlines. And adding more fields will make serializing the structs slower.

But we can expose the escape function as an utility, so that whoever likes our way of escaping newlines, can use it. A potential candidate for https://github.com/sayanarijit/xplr/issues/517